### PR TITLE
Teacher tool: block in set validator added

### PR DESCRIPTION
**Pull request by srietkerk**
This PR establishes the block in set validator. I thought the `blocksInSetExist` terminology made more sense than `blockSetExists` since we're not checking that a group of blocks exist. With this validator, we're checking that some kinds of blocks from a set exist, so I'm hoping `blocksInSetExist` is more descriptive. I'm up for changing it, though, if people feel strongly.

For its use for now, I plugged it into the `custom_function_call` validator plan we have in an example in our OneNote. It's a good example of how we'd want to use two different validators together where we want to make sure that a function is made and it is either called and the return function is accessed or it's just called. The `blocksInSetExist` validator is meant to handle this "or" scenario. 

Here are some other possible scenarios that I thought:
- Creating and manipulating a sprite (as in, a teacher wants their student to move the sprite, scale  the sprite, remove the sprite, add effects to the sprite, and it can be any of those)
- Creating and using a variable, although this might be better situated with the `blocksExist` validator
- Maybe something with extensions, like we ensure that an extension was installed and a certain number of blocks from that extension was used, but it doesn't matter which of those blocks were used
- Something with arrays similar to the sprite scenario that I explained above

### CODE REVIEW

File: pxtblocks/code-validation/validateBlocksInSetExist.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+namespace pxt.blocks {
+    // validates that a combination of blocks in the set satisfies the required count
+    // returns the blocks that make the validator pass
+    export function validateBlocksInSetExist({ usedBlocks, blocks, count }: {
+        usedBlocks: Blockly.Block[],
+        blocks: string[],
```
Comment (thsparks) [inline pxtblocks/code-validation/validateBlocksInSetExist.ts:6]: nit: perhaps a slightly more descriptive name could be helpful here, i.e. 'blockIdsToCheck'

File: pxtblocks/code-validation/validateBlocksInSetExist.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+namespace pxt.blocks {
+    // validates that a combination of blocks in the set satisfies the required count
+    // returns the blocks that make the validator pass
+    export function validateBlocksInSetExist({ usedBlocks, blocks, count }: {
```
Comment (thsparks) [inline pxtblocks/code-validation/validateBlocksInSetExist.ts:4]: Perhaps worth some discussion around whether or not we want to add a `requireUnique` flag here, which would make it so that each block in the set can only be counted once towards the desired total count. (i.e. instead of pushing all the blockInstances that match the id onto successfulBlocks, you push at most one, or zero if there are no matches).

Did you have any thoughts on this?

Comment (thsparks) [reply pxtblocks/code-validation/validateBlocksInSetExist.ts:4]: The kinds of scenarios where this would be helpful for teacher-tool are very edge-case-y, since you can (usually) just do it with blocksExist and a threshold. The only scenarios I can think of where that is not the case would be if you wanted to check something like "two different kinds of loops" AND a third criteria that's not directly tied to that, all in the same criteria for some reason.

But it could be quite useful for tutorials, which don't have the same check/threshold architecture.

**Review by thsparks — APPROVED** [review]
One nit and some food for thought, but all-in-all, LGTM!

File: pxtblocks/code-validation/validateBlocksInSetExist.ts (hunk 1 above)
Comment (srietkerk) [reply pxtblocks/code-validation/validateBlocksInSetExist.ts:6]: Quick question, if I change the name here, should I update the name in the validator plans and the interface definition as well? I know changing the parameter in the function is a bit more pertinent since that's where we'd interface with the validator elsewhere, and having the name be more descriptive is helpful when using the validator. I'm wondering if there should be consistency underneath as well.

Comment (thsparks) [reply pxtblocks/code-validation/validateBlocksInSetExist.ts:6]: I'd aim for consistency wherever possible :)

Comment (thsparks) [reply pxtblocks/code-validation/validateBlocksInSetExist.ts:6]: Actually I take it back in this case. I think blockIdsToCheck is too wordy for the actual validator plan. For that, I'd go with blocks (as you have it) or blockIds. The "toCheck" is a little less necessary, since there's no notion of the "usedBlocks" in that scenario, which is part of what causes the possible-confusion in this function.

File: pxtblocks/code-validation/validateBlocksInSetExist.ts (hunk 2 above)
Comment (thsparks) [reply pxtblocks/code-validation/validateBlocksInSetExist.ts:4]: Yeah that's reasonable. Some functionality, like requiring different counts for different blocks, is still done better with blocksExist, but it's possible that as this validator becomes more fleshed out, its functionality could become a superset of what blocksExist can do, then the blocksExist could head towards deprecation...but I wouldn't worry about that too much right now. For now, I think having the flag as an option but not using it in teacherTool is fine. If we want to use it later as a shorthand so we're not typing out a bunch of individual blocksExist checks, then that's great too. But no need for it to be a part of this change.

Comment (srietkerk) [reply pxtblocks/code-validation/validateBlocksInSetExist.ts:4]: Sounds good, I'll just add it as a parameter for the validator itself, then.
